traffic_ops/traffic_ops_golang/federation_resolvers/federation_resolvers.go

```diff
@@ -49,6 +52,18 @@ FROM federation_resolver
 LEFT OUTER JOIN type ON type.id = federation_resolver.type
 `
 
+const deleteQuery = `
+DELETE FROM federation_resolver
+WHERE federation_resolver.id = $1
+RETURNING federation_resolver.id,
+          federation_resolver.ip_address,
+          (
+          	SELECT type.name
+          	FROM type
+          	WHERE type.id = federation_resolver.type
+          ) AS type
+`
```

Comment:
Idk, I think it's standard to put constants and global variables at the top then subroutines below that, isn't it? I couldn't find a source for Go best-practices on the matter. I could add comments to them although they're un-exported if you think that'd help?

Reply:
I don't believe there is a conventional placement for constants and global variables in Go. Go is somewhat unique in that the order of declaration of top level identifiers mostly doesn't matter. The compiler determines the order to initialize variables and constants based on lexical dependencies (see: https://golang.org/ref/spec#Package_initialization). So if one depends on the other it can still appear first in a .go file and the compiler does the right thing.

This property of Go frees us from needing to declare things before using them at package scope and allows us to place them where it is most convenient for readability.

In my opinion, code is more readable when values are easy to find when reading the code that uses them. I also think code is more readable when identifiers are declared in the narrowest possible scope. Minimizing the portion of the code that has access to a value helps a maintainer know the possible impact of changing it.

In this case, if a future maintainer jumps into this code and sees the query string they will wonder where is it used and if they start with the function that runs the query they will wonder what the SQL looks like. It would be nice to have the two next to each other.

I am sure opinions vary on this subject, and that's OK, but those are the reasons behind my suggestion here. Feel free to take it or leave it as you see fit.

Reply:
I'd rather leave it. The layout of "global constants/variables first" matches the output format of GoDoc so I think that will make it easier for people to read, and also suggests that this is the manner that many find comfortable for reading it.